vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -32,23 +34,19 @@ def save_inventory_prep(find_key, child_keys, extra_keys)
     return find_key, child_keys, extra_keys, remove_keys
   end
 
-  def save_inventory(type, klass, parent, hash, deletes, new_records, record_index, record_index_columns, find_key, child_keys, remove_keys)
-    # Backup keys that cannot be written directly to the database
-    key_backup = backup_keys(hash, remove_keys)
-
+  def save_inventory(type, parent, hash, deletes = nil, new_records = nil, record_index = nil,
+                     record_index_columns = nil, find_key = nil)
     # Find the record, and update if found, else create it
     found = find_key.blank? ? parent.send(type) : self.save_inventory_record_index_fetch(record_index, record_index_columns, hash, find_key)
     if found.nil?
-      found = klass.new(hash)
-      new_records.nil? ? parent.send("#{type}=", found) : new_records << found
+      found = find_key ? parent.send(type).build(hash) : parent.send("build_#{type}", hash)
+      # TODO: these should not be necessary. Probably missing some parent.save calls
```

Comment:
1. good news: `save_inventory` only has 2 callers: `save_inventory_single` and `save_inventory_multi`.
2. `new_records` is only used in [`save_inventory_multi`](https://github.com/ManageIQ/manageiq/blob/master/vmdb/app/models/ems_refresh/save_inventory_helper.rb#L18) to add the child to the association.

```
class Parent < ActiveRecord::Base
  has_many :children
end

class Child < ActiveRecord::Base
  belongs_to :parent
end

class ParentTest < ActiveSupport::TestCase
    new_records = []
    p = Parent.new(name: "p1")

    new_records << p.children.build(name: "c1")
    new_records << p.children.build(name: "c2")

    # p.children.push(new_records) # works with or without this line
    p.save

    assert_equal 2, p.children.size
    assert_equal 2, p.children(true).count
    assert_equal 2, p.reload.children.count
end
```

We do have `has_many :through` and we reload the collections often (`children(true)`, but I don't think the commented out line should make a difference. The code is vast enough that I estimated it, and there may be a nuance in the real code that I missed.

@tenderlove any ideas why the commented out line would be required?